config/v1/stable.infrastructure.testsuite.yaml

```
@@ -214,3 +214,99 @@ tests:
+        spec:
+          platformSpec:
+            type: PowerVS
+        status:
+          platform: PowerVS
+          platformStatus:
+            powervs:
+              resourceGroup: resource-group
+      updated: |
+        apiVersion: config.openshift.io/v1
+        kind: Infrastructure
+        spec:
+          platformSpec:
+            type: PowerVS
+        status:
+          platform: PowerVS
+          platformStatus:
+            powervs:
+              resourceGroup: resource-group-should-not-accept-the-string-that-exceeds-max-length-set
+      expectedStatusError: "status.platformStatus.powervs.resourceGroup: Too long: may not be longer than 40"
+    - name: PowerVS platform status's resourceGroup should match the regex configured
+      initial: |
+        apiVersion: config.openshift.io/v1
+        kind: Infrastructure
+        spec:
+          platformSpec:
+            type: PowerVS
+        status:
+          platform: PowerVS
+          platformStatus:
+            powervs:
+              resourceGroup: resource-group
+      updated: |
+        apiVersion: config.openshift.io/v1
+        kind: Infrastructure
+        spec:
+          platformSpec:
+            type: PowerVS
+        status:
+          platform: PowerVS
+          platformStatus:
+            powervs:
+              resourceGroup: re$ource-group
+      expectedStatusError: "status.platformStatus.powervs.resourceGroup in body should match '^[a-zA-Z0-9-_ ]+$'"
+    - name: Should not be able to change PowerVS platform status's resourceGroup once it was set
+      initial: |
+        apiVersion: config.openshift.io/v1
+        kind: Infrastructure
+        spec:
+          platformSpec:
+            type: PowerVS
+        status:
+          platform: PowerVS
+          platformStatus:
+            powervs:
+              resourceGroup: resource-group
+      updated: |
+        apiVersion: config.openshift.io/v1
+        kind: Infrastructure
+        spec:
+          platformSpec:
+            type: PowerVS
+        status:
+          platform: PowerVS
+          platformStatus:
+            powervs:
+              resourceGroup: other-resource-group-name
+      expectedStatusError: "status.platformStatus.powervs: Invalid value: \"object\": resourceGroup is immutable once set"
+    - name: Should not be able to unset PowerVS platform status's resourceGroup once it was set
+      initial: |
+        apiVersion: config.openshift.io/v1
+        kind: Infrastructure
+        spec:
+          platformSpec:
+            type: PowerVS
+        status:
+          platform: PowerVS
+          platformStatus:
+            powervs:
+              resourceGroup: resource-group
+      updated: |
+        apiVersion: config.openshift.io/v1
+        kind: Infrastructure
+        spec:
+          platformSpec:
+            type: PowerVS
+        status:
+          platform: PowerVS
+          platformStatus:
+            powervs:
+              resourceGroup:
```

Do you get the same result if you change this to the following?
```suggestion
            powervs: {}
```

---

No getting nil.

---

Try instead 
```
platformStatus:
  powerVS:
    regionName: foo
    resourceGroup: bar
```
And then remove the resourceGroup entry completely, then I think things should work

---

If we want `powervs: {}` to return statusError requires validation rule to be added in `PlatformStatus` struct.

Made the changes you suggested and test case passed. PTAL!

---

Yep agreed, don't want to scope creep too much though, so lets leave as is for now, thanks